tests/unittests/web/jwtgen_test.py

```diff
@@ -0,0 +1,106 @@
+import pytest
+from unittest.mock import Mock, patch
+from datetime import datetime
+
+import jwt
+
+from nav.web.jwtgen import generate_access_token, generate_refresh_token
+
+
+class TestTokenGeneration:
+    """Tests behaviour that should be identical for both access and refresh token generation"""
+
+    @pytest.mark.parametrize("func", [generate_access_token, generate_refresh_token])
+    def test_nbf_should_be_in_the_past(self, func):
+        encoded_token = func()
+        data = jwt.decode(encoded_token, options={'verify_signature': False})
+        assert data['nbf'] < datetime.now().timestamp()
+
+    @pytest.mark.parametrize("func", [generate_access_token, generate_refresh_token])
+    def test_exp_should_be_in_the_future(self, func):
+        encoded_token = func()
+        data = jwt.decode(encoded_token, options={'verify_signature': False})
+        assert data['exp'] > datetime.now().timestamp()
+
+    @pytest.mark.parametrize("func", [generate_access_token, generate_refresh_token])
+    def test_iat_should_be_in_the_past(self, func):
+        encoded_token = func()
+        data = jwt.decode(encoded_token, options={'verify_signature': False})
+        assert data['iat'] < datetime.now().timestamp()
+
+    @pytest.mark.parametrize("func", [generate_access_token, generate_refresh_token])
+    def test_aud_should_match_name_from_jwt_conf(self, func, nav_name):
+        encoded_token = func()
+        data = jwt.decode(encoded_token, options={'verify_signature': False})
+        assert data['aud'] == nav_name
+
+    @pytest.mark.parametrize("func", [generate_access_token, generate_refresh_token])
+    def test_iss_should_match_name_from_jwt_conf(self, func, nav_name):
+        encoded_token = func()
+        data = jwt.decode(encoded_token, options={'verify_signature': False})
+        assert data['iss'] == nav_name
+
+
+class TestGenerateAccessToken:
+    def test_token_type_should_be_access_token(self):
+        encoded_token = generate_access_token()
+        data = jwt.decode(encoded_token, options={'verify_signature': False})
+        assert data['token_type'] == "access_token"
+
+
+class TestGenerateRefreshToken:
+    def test_token_type_should_be_refresh_token(self):
+        encoded_token = generate_refresh_token()
+        data = jwt.decode(encoded_token, options={'verify_signature': False})
+        assert data['token_type'] == "refresh_token"
+
+
+@pytest.fixture(scope="module", autouse=True)
```

Comment:
Just curious why you picked "module" scope for this fixture?

Reply:
All tests need this (hence "autouse"), so since all tests use it I set "module" scope so it only runs once. My intention is that this can mock `JWTConf` for all tests by only running once